### DISCUSSION TITLE
Welcome to tsParticles new ideas discussion

### DISCUSSION THREAD

**Discussion by matteobruni** [matteobruni/tsparticles#1118] [Ideas]
Hi everyone.

This discussion will be used to propose and discuss any new idea to improve in any way the product.

Some of them can be found in the `Issues` section but I think we can discuss them here before creating an issue, to find if the idea is valid and not already done.

**Comment by cogentgroup** [comment]
We love tsParticles but very new to code.  Your work is a great discovery.  Thank you for your vision and work!  

Question:  How can we attach words (similar to the image) each particle and have it change size relative to its position of the particle.  I've attached a couple of images that illustrate the kind of effect we're looking for.  Way back in the late 90's the Acura.com website had something similar where you could drag the particle (as I remember) and move the particle / word...  dragging the particle / word around isn't necessary but we'd like to link the words with the particles if possible. 



![Particle_Word_Connection_2](https://user-images.githubusercontent.com/9917656/105117585-fad1b980-5a89-11eb-9ff4-66969ebaa4c9.jpg)
![Particle_Word_Connection_1](https://user-images.githubusercontent.com/9917656/105117587-fb6a5000-5a89-11eb-8018-a8b8c756dc4b.jpg)

**Reply by matteobruni** [reply]
Hi @cogentgroup,

This feature could be achieved but it's not an easy one. Actually, there are no shapes that support that kind of shape (centered circle and text near) but a custom shape can be created.
You can see some external custom shapes created here: https://github.com/matteobruni/tsparticles/tree/master/shapes
If you need it shortly I'm sorry I can't work on it, I'm currently trying to complete the `2.0` version at the moment and it will be a huge release

**Comment by cogentgroup** [comment]
Matteo,

 

I appreciate your quick response.  

 

I remember seeing this 20 years ago on the Acura.com site and thought wow… that’s very advanced for the technology of that time.  

 

I’m guessing it was not js…  maybe someone else will know what script that could have been.

 

We will live without this feature for now but would love to discuss how this could be done after you release 2.0!  

 

Your scripts are amazing and we thank you for the great work and effort you’ve put into this.  

 

Can’t wait to see 2.0!

 

Thank you again.
